### DISCUSSION TITLE
Preview window

### DISCUSSION THREAD

**Discussion by zethon** [dannagle/PacketSender#253] [Ideas]
I use this tool a lot and love it. 

However one thing that annoys me is that there's no easy to view the data in the "ASCII/Hex" `QLineEdit` boxes. Yes, it's there in the two `QLineEdit` controls but for things like HTTP requests/responses or other large'ish data, it's difficult to see what's there.

I'm working on a `Preview` button that will launch a read-only modal dialog with only a "Close" button. I could see it eventually allow editing, copying, etc, etc... 

My question is where is the "truth" of the "Send" button stored? It looks like it's in the `ui->packetHexEdit` control since that's where "Load File" puts it. So, my Preview button would do something like:

* get the text from `ui->packetHexEdit`
* convert it to ASCII (converting the `\r` and `\n` characters to actual whitespace)
* open the dialog
* populate the text

Or is there a more "truthy" object that stores the data?

**Comment by dannagle** [comment]
Yes, the hex field holds the truth. Throughout the code, Packet Sender reads hex to convert to ascii.

This is a neat idea. However, before you go too far into this, you may want to switch to the development branch. The next major version of Packet Sender (version 8) has HTTP(S) GET/POST as first-class citizens with controls to help you craft params.

**Comment by dannagle** [comment]
Adding a reference to #254 in this discussion.

**Comment by dannagle** [comment]
This is a neat addition. I tweaked the preview dialog to allow editing if double-click ASCII. Before moving this feature to master, I think double-click HEX should preview hex and double-click ascii should preview ascii. Both could support editing their lines in that way. That would feel more intuitive.

**Comment by zethon** [comment]
It would be neat if the hex option launched a hex editor instead of a plaintext editor. I've yet to find a decent hex editting Qt component though.
